src/utils.js

```diff
@@ -208,6 +208,17 @@ _.toArray = function(iterable) {
     return _.values(iterable);
 };
 
+_.keys = function(obj) {
+    var results = [];
+    if (obj === null) {
+        return results;
+    }
+    _.each(obj, function(value, key) {
+        results[results.length] = key;
```

Comment:
can we not use `results.push`?

Reply:
I was just mirroring the `each`-based implementation of `_.values()` right below for consistency, not particularly wedded to this weird implementation that probably has its roots in some ancient runtime optimization...